src/libsync/bulkpropagatorjob.cpp

```diff
@@ -0,0 +1,690 @@
+{
+    return reply.value(headerName).toString().toLatin1();
+}
+
+}
+
+namespace OCC {
+
+BulkPropagatorJob::BulkPropagatorJob(OwncloudPropagator *propagator,
+                                     const std::deque<SyncFileItemPtr> &items)
+    : PropagatorJob(propagator)
+    , _items(items)
+{
+    _uploadFileParameters.reserve(100);
+}
+
+bool BulkPropagatorJob::scheduleSelfOrChild()
+{
+    if (_items.empty()) {
+        return false;
+    }
+
+    {
+        auto bulkUploadUrl = Utility::concatUrlPath(propagator()->account()->url(), QStringLiteral("/remote.php/dav/bulk"));
+    }
+
+    _state = Running;
+    for(int i = 0; i < 100 && !_items.empty(); ++i) {
+        auto currentItem = _items.front();
+        _items.pop_front();
+        _pendingChecksumFiles.insert(currentItem->_file);
+        QMetaObject::invokeMethod(this, [this, currentItem] () {
+            UploadFileInfo fileToUpload;
+            fileToUpload._file = currentItem->_file;
+            fileToUpload._size = currentItem->_size;
+            fileToUpload._path = propagator()->fullLocalPath(fileToUpload._file);
+            startUploadFile(currentItem, fileToUpload);
+        }); // We could be in a different thread (neon jobs)
+    }
+
+    return _items.empty() && _uploadFileParameters.empty();
+}
+
+PropagatorJob::JobParallelism BulkPropagatorJob::parallelism()
+{
+    return PropagatorJob::JobParallelism::WaitForFinished;
+}
+
+void BulkPropagatorJob::startUploadFile(SyncFileItemPtr item, UploadFileInfo fileToUpload)
+{
+    if (propagator()->_abortRequested) {
+        return;
+    }
+
+    // Check if the specific file can be accessed
+    if (propagator()->hasCaseClashAccessibilityProblem(fileToUpload._file)) {
+        done(item, SyncFileItem::NormalError, tr("File %1 cannot be uploaded because another file with the same name, differing only in case, exists").arg(QDir::toNativeSeparators(item->_file)));
+        return;
+    }
+
+    return slotComputeTransmissionChecksum(item, fileToUpload);
+}
+
+void BulkPropagatorJob::doStartUpload(SyncFileItemPtr item,
+                                      UploadFileInfo fileToUpload,
+                                      QByteArray transmissionChecksumHeader)
+{
+    if (propagator()->_abortRequested) {
+        return;
+    }
+
+    // write the checksum in the database, so if the POST is sent
+    // to the server, but the connection drops before we get the etag, we can check the checksum
+    // in reconcile (issue #5106)
+    SyncJournalDb::UploadInfo pi;
+    pi._valid = true;
+    pi._chunk = 0;
+    pi._transferid = 0; // We set a null transfer id because it is not chunked.
+    pi._modtime = item->_modtime;
+    pi._errorCount = 0;
+    pi._contentChecksum = item->_checksumHeader;
+    pi._size = item->_size;
+    propagator()->_journal->setUploadInfo(item->_file, pi);
+    propagator()->_journal->commit("Upload info");
+
+    qint64 fileSize = fileToUpload._size;
```

Comment:
@mgallien Are these extra variables `fileSize`, `path`, `fileName` even needed?

Reply:
not at all
going to clean up this